src/mono/sample/wasm/browser-threads-minimal/WebWorker.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.JavaScript;
+using System.Threading;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace System.Runtime.InteropServices.JavaScript
+{
+    public partial class WebWorker
+    {
+        public static Task<T> Run<T>(Func<Task<T>> body)
+        {
+            var tcs = new TaskCompletionSource<T>();
+            var t = new Thread(() => {
+                Task<T> res = body();
+                res.ContinueWith((t) => {
```

Comment:
This is weird, is the goal to register a continuation on this thread and then let the thread die? If so, why?

Reply:
Yes, this is confusing. I added comments
```
                // This code is exiting thread main() before all promises are resolved.
                // the continuation is executed by setTimeout() callback of the thread. 
```